### PR TITLE
Update some vendoring with tags

### PR DESCRIPTION
**Pull request by vdemeester**
Following the [vendoring policy](https://github.com/docker/docker/blob/master/VENDORING.md), I tried to update a bit of vendoring with `tagged` versions 👼.
- [ ] Make sure the actual changes do not add regression (otherwise I'll add back the hash and add a comment)
- [ ] Few `docker/*` project need to have tag version (added a `FIXME` and created some issues about it in the corresponding repositories)
- Some tagged version imply probably too much changes (`github.com/go-ini/ini` +2500 or `github.com/tinylib/msgp`), I added a comment to update them but I feel it's too late for it to be update for `v1.12.0`

I'm putting `1.12.0` milestone on it but feel free to remove it if you feel it's not needed or risky :wink:.

/cc @tiborvass @icecrime @anusha-ragunathan @thaJeztah 

Signed-off-by: Vincent Demeester vincent@sbr.pm

### CODE REVIEW

File: hack/vendor.sh [hunk 1]
```diff
@@ -99,7 +99,7 @@ clone git github.com/docker/notary v0.3.0
 
 clone git google.golang.org/grpc ab0be5212fb225475f2087566eded7da5d727960 https://github.com/grpc/grpc-go.git
 clone git github.com/miekg/pkcs11 df8ae6ca730422dba20c768ff38ef7d79077a59f
-clone git github.com/docker/go v1.5.1-1-1-gbaf439e
```
Comment (vdemeester) [inline hack/vendor.sh]: I could not find this tag, but `v1.5.1-1` seems to be the same :angel:

**Comment by thaJeztah** [comment]
sgtm; but not sure indeed if we should *bump* versions during the RC unless there's a change that's needed for 1.12.

**Comment by vdemeester** [comment]
@thaJeztah yep wasn't sure either. At 'worst' I can just add changes for updates that do not change any code 😇

**Comment by LK4D4** [comment]
Let's keep it for 1.13.

**Comment by anusha-ragunathan** [comment]
I :heart: this change

**Comment by vdemeester** [comment]
I'll close this for now and will re-open and update when 1.12.0 is released) :wink:

**Pull request closed without merging**